yjit/src/invariants.rs

```diff
@@ -15,10 +15,25 @@ use std::collections::HashMap;
 // assume_single_ractor_mode(jit)
 // assume_stable_global_constant_state(jit);
 
+struct MethodLookupDependency {
+    block: BlockRef,
+    mid: ID
+}
+
 /// Used to track all of the various block references that contain assumptions
 /// about the state of the virtual machine.
 pub struct Invariants {
-    bops: HashMap<(RedefinitionFlag, ruby_basic_operators), Vec<BlockRef>>
+    /// Tracks block assumptions about the stability of a basic operator on a
+    /// given class.
+    basic_operators: HashMap<(RedefinitionFlag, ruby_basic_operators), Vec<BlockRef>>,
+
+    /// Tracks block assumptions about callable method entry validity.
+    cme_validity: HashMap<*const rb_callable_method_entry_t, Vec<BlockRef>>,
```

Comment:
You can only use `rb_callable_method_entry_t` as a key in a hash map if we're sure that the pointer is not movable by the Ruby GC. Is this the case? If so, that should be commented.